### ISSUE TITLE
Need to support "aspose"

### ISSUE THREAD

**Issue by KevinRodriguez777** [Nuitka/Nuitka#2713]
> python -m nuitka --version
```
2.0.2
Commercial: None
Python: 3.8.10 (tags/v3.8.10:3d8993a, May  3 2021, 11:48:03) [MSC v.1928 64 bit (AMD64)]
Flavor: CPython Official
Executable: C:\Users\Divine\Desktop\PyDis\venv\Scripts\python.exe
OS: Windows
Arch: x86_64
WindowsRelease: 10
```
- How did you install Nuitka and Python
cpython / pip install

  > python -m pip freeze

you'll need to pip install `Aspose.Email-for-Python-via-NET`

- Also supply a Short, Self Contained, Correct, Example
```
from aspose.email.mapi import MapiMessage
# from aspose.email import mapi

def main():
    msg = mapi.MapiMessage.load(r"testing.eml")
    print("Subject:", msg.subject)
    print("From:", msg.sender_email_address)


if __name__ == "__main__":
    main()
```
with the commented code available it will work when compiled
- Provide in your issue the Nuitka options used
`nuitka --standalone main.py`

**Comment by kayhayen** [comment]
I am assuming I will need yaml config for this one too, right?

**Comment by kayhayen** [comment]
Seems this is what is needed:

```
- module-name: 'aspose.email'
  dlls:
    - from_filenames:
        prefixes:
          - 'email'
          - 'pycore'
          - 'pygc'
          - 'pyreflection'
        suffixes:
          - 'pyd'
      when: 'win32'
    - from_filenames:
        relative_path: 'assemblies/email'
        prefixes:
          - 'Aspose'
          - 'System'
          - 'WrpInterop'
      when: 'win32'
    - from_filenames:
        relative_path: 'assemblies/pycore'
        prefixes:
          - 'WrpInterop'
      when: 'win32'
    - from_filenames:
        relative_path: 'assemblies/pygc'
        prefixes:
          - 'Aspose'
      when: 'win32'
    - from_filenames:
        relative_path: 'assemblies/pyreflection'
        prefixes:
          - 'WrpInterop'
      when: 'win32'
    - from_filenames:
        relative_path: 'netcore/netcore3.1'
        prefixes:
          - 'aspose'
          - 'System'
          - 'api-ms-win'
          - 'Microsoft'
          - 'mscor'
          - 'ucrtbase'
          - 'WindowsBase'
          - 'netstandard'
          - 'host'
          - 'dbgshim'
          - 'clr'
          - 'coreclr'
      when: 'win32'
```

**Comment by kayhayen** [comment]
That looks horribly wrong:

```
- module-name: 'aspose.email'
  dlls:
    - from_filenames:
        prefixes:
          - 'email'
          - 'pycore'
          - 'pygc'
          - 'pyreflection'
        suffixes:
          - 'pyd'
      when: 'win32'
```

Given that `email` is the one making the trouble, I wonder how that happens, but that looks like a poor DLL configuration or a very poor upstream. I think "email" will slash will "email", maybe therefore not being something to import from anymore?

**Comment by kayhayen** [comment]
That's how these seem to be done correctly, not getting all you had there:

```
- module-name: 'aspose.email'
  implicit-imports:
    - depends:
        - 'aspose.pyreflection'
        - 'aspose.pygc'
```

DLLs with "pyd" suffix should *only* be used when it is *not* an importable module. Doing them as DLLs is defeating the ability of Nuitka to import things, as it will not know that it's an extension module.

**Comment by KevinRodriguez777** [comment]
i had initially tried that, but it ended up crashing the exe on startup. the YAML i sent is what i finally got nuitka to build and run without crashing at startup, but rather give an runtime error

**Comment by KevinRodriguez777** [comment]
the runtime / import error being 
```
Traceback (most recent call last):
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 635, in _load_backward_compatible
KeyError: 'aspose.pyreflection'

The above exception was the direct cause of the following exception:

ImportError: Unable to import module dependencies. Cannot import the aspose.pyreflection module. The module not found or errors occurred while initializing it.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Divine\Desktop\PyDis\MAIN~1.DIS\main.py", line 2, in <module>
ImportError: One or more errors occurred while loading the module 'aspose.email' (-1009)
``` without including the pyd

**Comment by KevinRodriguez777** [comment]
never mind, i'm wrong, seems like it was a misconfiguration of the yaml, works via 
```yaml
- module-name: 'aspose.email'
  implicit-imports:
    - depends:
        - 'aspose.pyreflection'
        - 'aspose.pygc'
  dlls:
    - from_filenames:
        relative_path: 'assemblies/email'
        prefixes:
          - 'Aspose'
          - 'System'
          - 'WrpInterop'
      when: 'win32'
    - from_filenames:
        relative_path: 'assemblies/pycore'
        prefixes:
          - 'WrpInterop'
      when: 'win32'
    - from_filenames:
        relative_path: 'assemblies/pygc'
        prefixes:
          - 'Aspose'
      when: 'win32'
    - from_filenames:
        relative_path: 'assemblies/pyreflection'
        prefixes:
          - 'WrpInterop'
      when: 'win32'
    - from_filenames:
        relative_path: 'netcore/netcore3.1'
        prefixes:
          - 'aspose'
          - 'System'
          - 'api-ms-win'
          - 'Microsoft'
          - 'mscor'
          - 'ucrtbase'
          - 'WindowsBase'
          - 'netstandard'
          - 'host'
          - 'dbgshim'
          - 'clr'
          - 'coreclr'
      when: 'win32'```

**Comment by kayhayen** [comment]
My hopes have been shattered though, it seems the issue persists, email is both a directory and an extension module, I was thinking that might play a role.

**Comment by kayhayen** [comment]
So, I found the probably bad thing:

```
foreach_file(AsposeModuleLoader.aspose_native_lib_dir,
             _platform_suffixes, lambda name, path: __all__.append(name))

sys.meta_path.append(AsposeModuleFinder())
```

**Comment by kayhayen** [comment]
I don't know if this is doable, they have their own source file loader, as as meta path based loader, and I guess, aspose.email is supposed to be loaded by that already, and not by Nuitka's meta path based loader.

This is so deeply broken, I cannot currently think of a way to solve it. I will try and exclude the extension modules from loading by Nuitka, and see what that does. I tried rather than appending to sys.meta_path, to insert to 0, but then nothing loads anymore, so maybe that's going to help some.

**Comment by kayhayen** [comment]
Turns out, it is doable, since we can let their code do the extension module loading, and there is no Python code in there, so whatever it is that they do with source code, that's probably generated one.

What I did seems not limited to `aspose.email` but maybe goes beyond that.

**Comment by kayhayen** [comment]
It seems, my backport to 2.0.4 hotfix has failed, that's not going to work, unless I split it up into a unoptimal part, where all Python versions extension modules are included, and one where in 2.1 that gets improved. However, that doesn't make much sense, I guess we rather focus on having 2.1 with this released.

**Comment by kayhayen** [comment]
This is part of the 2.1 release I just made. For some reason, Linux and macOS trigger assertions when loading a DLL, this is apparently not covered yet.
